packages/react-vapor/src/components/unsavedChangeModal/test/UnsavedChangeModalProvider.spec.tsx

```diff
@@ -0,0 +1,155 @@
+
+            const heavyModal: any = heavyUnsavedChangesModalProvider.find(ModalCompositeConnected);
+            heavyModal.props().modalFooterChildren.props.onClick();
+            heavyUnsavedChangesModalProvider.update();
+            expect(heavyUnsavedChangesModalProvider.childAt(1).prop('isOpen')).toBeTruthy();
+        });
+        describe('PromptBefore function', () => {
+            it('should return false', () => {
+                unsavedChangesModalProvider = shallowMountUnsavedModalProvider(true);
+                unsavedChangesModalProvider
+                    .childAt(0)
+                    .props()
+                    .modalFooterChildren.props.onClick();
+                expect(regularClickActionSpy).not.toHaveBeenCalled();
+            });
+        });
+        it('should use the callBack function in the promptBefore function when confirming the exit', () => {
+            heavyUnsavedChangesModalProvider = heavilyMountUnsavedModalProvider(true);
+
+            const modal: any = heavyUnsavedChangesModalProvider.find(ModalCompositeConnected);
+            modal.props().modalFooterChildren.props.onClick();
+            heavyUnsavedChangesModalProvider.update();
+            const exitWithoutSavingButton: any = heavyUnsavedChangesModalProvider
+                .childAt(1)
+                .find('Button')
+                .first();
+            exitWithoutSavingButton.prop('onClick')();
+
+            expect(promptBeforeClickActionSpy).toHaveBeenCalled();
+        });
+        it('should not use the callBack function in the promptBefore function when cancelling the exit', () => {
+            heavyUnsavedChangesModalProvider = heavilyMountUnsavedModalProvider(true);
+
+            const modal: any = heavyUnsavedChangesModalProvider.find(ModalCompositeConnected);
+            modal.props().modalFooterChildren.props.onClick();
+            heavyUnsavedChangesModalProvider.update();
+            const exitWithoutSavingButton: any = heavyUnsavedChangesModalProvider
+                .childAt(1)
+                .find('Button')
+                .last();
+            exitWithoutSavingButton.prop('onClick')();
+
+            expect(promptBeforeClickActionSpy).not.toHaveBeenCalled();
+        });
+    });
+
+    describe('when the modal is not dirty', () => {
+        it('should not open the UnsavedChangesModal to interrupt the leaving action', () => {
+            heavyUnsavedChangesModalProvider = heavilyMountUnsavedModalProvider(false);
+            expect(heavyUnsavedChangesModalProvider.childAt(1).prop('isOpen')).toBeFalsy();
+
+            const heavyModal: any = heavyUnsavedChangesModalProvider.find(ModalCompositeConnected);
+            heavyModal.props().modalFooterChildren.props.onClick();
+            heavyUnsavedChangesModalProvider.update();
+            expect(heavyUnsavedChangesModalProvider.childAt(1).prop('isOpen')).toBeFalsy();
+        });
+        it('should return true', () => {
```

Comment:
🤔 not sure what returns true just with the name, I think you could add "promptBefore should return true"

Reply:
this It description struck me as very incomplete and I was about to change it anyway. I will probably take your suggestion. :+1: